src/tests/multihost/sssd/testlib/common/utils.py

```diff
@@ -1020,6 +1021,87 @@ def remove_service_principals(self, spn_list):
             except subprocess.CalledProcessError:
                 pytest.fail("setspn failed to delete %s SPN" % (short_entry))
 
+    @staticmethod
+    def skip_package_version(
+            multihost_host: object,
+            min_versions: list = None,
+            max_versions: list = None,
+            package: str = "sssd",
+            skip_missing: bool = True,
+            fail_missing: bool = False
+        ):
+        """ Skip test if the package version is not between min_versions and max_versions
+
+        If multiple min versions are defined it is expected that the fix was introduced
+        to different branches at once and the comparison will compare to multiple min versions.
+        For version comparison purposes <major>.<minor> is considered one branch.
+        For branches where min version is not defined, but they fit between other min versions it is
```

Comment:
Let me explain on example:
Lets say that some bug was fixed in  sssd 2.9.2 but fix was backported to 2.7 branch as well in 2.7.4-5.
min_versions=["2.7.4-5", "2.9.2"]
Lets say that it finds installed sssd 2.9.1. 
It fits sssd-2-9 (<major>.<minor>) so the detected version is not compared to any other min_version and as it is lower that this mine version test will be skipped.

Lets say that test finds 2.8.3 as installed sssd.
We do not have defined any min version for 2.8  branch and 2.8.3 is higher than 2.7.4-5 so it will consider the test valid there and execute. If we wanted to skip 2.8 we would need define min_versions=["2.7.4-5", "2.8.999" , "2.9.2"].

The idea is that we can track minimal (package) version across multiple branches as the same fix could have been backported to different rhel versions. The max versions are similar concept for tests that will no longer by meaningful after some sssd change.

If we add this info to newly introduced tests we can just execute master.

Reply:
First of all, let me apologize for the long delayed response here.

> ```
> Let me explain on example:
> Lets say that some bug was fixed in  sssd 2.9.2 but fix was backported to 2.7 branch as well in 2.7.4-5.
> min_versions=["2.7.4-5", "2.9.2"]
> Lets say that it finds installed sssd 2.9.1. 
> It fits sssd-2-9 (<major>.<minor>) so the detected version is not compared to any other min_version and as it is lower that this mine version test will be skipped.
> 
> Lets say that test finds 2.8.3 as installed sssd.
> We do not have defined any min version for 2.8  branch and 2.8.3 is higher than 2.7.4-5 so it will consider the test valid there and execute. If we wanted to skip 2.8 we would need define min_versions=["2.7.4-5", "2.8.999" , "2.9.2"].

Thanks for the explanation.  It makes more sense to me now but, still seems unintuitive (to me).    I suppose you can't rely on adding every maj.min version where it should run because we'd have to add to this every time we release a new maj.min branch.    

I'd suggest two things:

1. Try to reword:

`For branches where min version is not defined, but they fit between other min versions`

to something like:

`If run for a case where the SSSD <major>.<minor> version is not defined, but is between the listed versions`

2. Add an example like you listed above in the comments

> 
> The idea is that we can track minimal (package) version across multiple branches as the same fix could have been backported to different rhel versions. The max versions are similar concept for tests that will no longer by meaningful after some sssd change.
> 
> If we add this info to newly introduced tests we can just execute master. 
> ```

I thought we agreed to include tests in the relevant branches alongside the features/fixes and where ever those are backported.   This sounds like we might not be doing that.  Would this be more for identifying tests within a <major>-<minor> branch where they could have been added much later than the feature/fix?